core/src/main/java/org/infinispan/distexec/spi/DefaultDistributedTaskLifecycle.java

```diff
@@ -9,7 +9,7 @@
 public class DefaultDistributedTaskLifecycle implements DistributedTaskLifecycle {
 
    @Override
-   public <T, K, V> void onPreExecute(Callable<T> task, Cache<K, V> inputCache) {
+   public <T, K, V> void onPreExecute(Callable<T> task, Cache<K, V> inputCache, K... inputKeys) {
```

Comment:
In general the code looks fine to me. 

The only comment that I have is that inputKeys should be a list (instead of array). This would simplify (a little bit but still) the code in `DefaultExecutorService`. Moreover, you should pass `Collections.unmodifiableList()` there.

Good work @SetoKaiba !